### PR TITLE
[ #99 ] Improved silence removal

### PR DESCRIPTION
**Pull request by Jlisowskyy**
Things done:
- Modified function detecting silence
- Added function cutting audio chunks to smaller chunks and reassembling the full cleaned chunk back together
- Added test for functionalities
- Added transform function to wav iterators
- Modified Flatten Wav iterator to flatten the audio on construction

results:
image
image
image

### CODE REVIEW

File: src/pipeline/wav.py [hunk 1]
```diff
@@ -10,7 +10,7 @@
 import wave
 from abc import ABC, abstractmethod
 from collections.abc import Generator
-from typing import Iterator, Union
+from typing import Iterator, Union, Callable
```
Comment (al-gor1thm) [inline src/pipeline/wav.py]: `typing` is deprecated
`Union[foo, bar]` is arguably clunkier compared to `foo | bar`

**Review by al-gor1thm — APPROVED** [review]
Looks good

**Review by kryczkal — APPROVED** [review]
LGTM

File: src/pipeline/wav.py (hunk 1 above)
Comment (Jlisowskyy) [reply src/pipeline/wav.py]: Thanks for the fix!

**Pull request merged**